packages/client/hmi-client/src/components/model/petrinet/tera-parameter-table-v2.vue

```diff
@@ -49,41 +99,99 @@
 			:open-value-config="!!matrixModalId"
 			@close-modal="matrixModalId = ''"
 			@update-cell-value="
-				emit('update-parameter', {
-					id: $event.variableName,
-					distribution: { type: DistributionType.Constant, parameters: { value: $event.newValue } }
-				})
+				emit('update-parameters', [
+					{
+						id: $event.variableName,
+						distribution: {
+							type: DistributionType.Constant,
+							parameters: { value: $event.newValue }
+						}
+					}
+				])
 			"
 		/>
 	</Teleport>
 </template>
 
 <script setup lang="ts">
-import { ModelConfiguration } from '@/types/Types';
-import { getParameters } from '@/services/model-configurations';
+import { ModelConfiguration, ModelDistribution } from '@/types/Types';
+import { getParameterDistribution, getParameters } from '@/services/model-configurations';
 import { StratifiedMatrix } from '@/types/Model';
-import { ref } from 'vue';
+import { computed, ref } from 'vue';
 import { collapseParameters, isStratifiedModel } from '@/model-representation/mira/mira';
 import { MiraModel, MiraTemplateParams } from '@/model-representation/mira/mira-common';
 import Accordion from 'primevue/accordion';
 import AccordionTab from 'primevue/accordiontab';
 import Button from 'primevue/button';
 import Divider from 'primevue/divider';
-import { DistributionType } from '@/services/distribution';
-import TeraStratifiedMatrixModal from './model-configurations/tera-stratified-matrix-modal.vue';
+import { DistributionType, distributionTypeOptions } from '@/services/distribution';
+import InputNumber from 'primevue/inputnumber';
+import Dropdown from 'primevue/dropdown';
+import Checkbox from 'primevue/checkbox';
 import TeraParameterEntry from './tera-parameter-entry.vue';
+import TeraStratifiedMatrixModal from './model-configurations/tera-stratified-matrix-modal.vue';
 
 const props = defineProps<{
 	modelConfiguration: ModelConfiguration;
 	mmt: MiraModel;
 	mmtParams: MiraTemplateParams;
 }>();
 
-const emit = defineEmits(['update-parameter', 'update-source', 'delete-distribution']);
+const emit = defineEmits(['update-parameters', 'update-source']);
 
 const isStratified = isStratifiedModel(props.mmt);
 
+const isAddingUncertainty = ref(false);
+const uncertaintyType = ref(DistributionType.Uniform);
+const uncertaintyPercentage = ref(10);
+const selectedParameters = ref<string[]>([]);
+
+const numParameters = computed(() => Object.keys(props.mmt.parameters).length);
+
 const matrixModalId = ref('');
+
+const onAddUncertainty = () => {
+	const selected = Object.keys(props.mmt.parameters);
```

Comment:
You should filter to select only the constant and not the distributions.

Reply:
easily yes, will do

Reply:
this has been updated, we can now only select constants and not distributions